src/sentry/sentry_apps/external_requests/select_requester.py

```diff
@@ -41,66 +49,83 @@ class SelectRequester:
     dependent_data: str | None = field(default=None)
 
     def run(self) -> SelectRequesterResult:
+
         response: list[dict[str, str]] = []
         url = None
-        try:
-            url = self._build_url()
-            body = safe_urlread(
-                send_and_save_sentry_app_request(
-                    url,
-                    self.sentry_app,
-                    self.install.organization_id,
-                    "select_options.requested",
-                    headers=self._build_headers(),
-                )
-            )
 
-            response = json.loads(body)
-        except Exception as e:
-            extra = {
+        with SentryAppInteractionEvent(
+            operation_type=SentryAppInteractionType.EXTERNAL_REQUEST,
+            event_type=SentryAppEventType.SELECT_OPTIONS_REQUESTED,
+        ).capture() as lifecycle:
+            extras: dict[str, Any] = {
                 "sentry_app_slug": self.sentry_app.slug,
                 "install_uuid": self.install.uuid,
                 "project_slug": self.project_slug,
             }
+            lifecycle.add_extras(extras)
+
+            try:
+                url = self._build_url()
+                extras.update({"url": url})
+
+                body = safe_urlread(
+                    send_and_save_sentry_app_request(
+                        url,
+                        self.sentry_app,
+                        self.install.organization_id,
+                        SentryAppEventType.SELECT_OPTIONS_REQUESTED,
+                        headers=self._build_headers(),
+                    )
+                )
+
+                response = json.loads(body)
+                extras.update({"response": response})
+            except RequestException as e:
+                halt_reason = f"{SentryAppEventType.SELECT_OPTIONS_REQUESTED}.{SentryAppExternalRequestHaltReason.BAD_RESPONSE}"
+                lifecycle.record_halt(halt_reason=e, extra={"reason": halt_reason, **extras})
 
-            if not url:
-                extra.update(
-                    {
-                        "uri": self.uri,
-                        "dependent_data": self.dependent_data,
-                        "webhook_url": self.sentry_app.webhook_url,
-                    }
+                raise SentryAppIntegratorError(
+                    message=f"Something went wrong while getting options for Select FormField from {self.sentry_app.slug}",
+                    webhook_context={"error_type": halt_reason, **extras},
+                    status_code=500,
+                ) from e
+
+            except Exception as e:
+                failure_reason = f"{SentryAppEventType.SELECT_OPTIONS_REQUESTED}.{SentryAppExternalRequestFailureReason.UNEXPECTED_ERROR}"
```

Comment:
if a bunch of these have the same `SentryAppEventType.SELECT_OPTIONS_REQUESTED` at the beginning, you could generate this string early on as a constant and add the failure reason at the end with `str.format()`, something like
```
FAILURE_REASON_BASE = f"{SentryAppEventType.SELECT_OPTIONS_REQUESTED} {{}}"
...
failure_reason = FAILURE_REASON_BASE.format(SentryAppExternalRequestFailureReason.UNEXPECTED_ERROR)
```